tests/Fixtures/TestBundle/State/UserResetPasswordProcessor.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Tests\Fixtures\TestBundle\State;
+
+use ApiPlatform\State\ProcessorInterface;
+use ApiPlatform\Tests\Fixtures\TestBundle\Dto\UserResetPasswordDto;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+class UserResetPasswordProcessor implements ProcessorInterface
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function resumable(?string $operationName = null, array $context = []): bool
+    {
+        return false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function supports($data, array $identifiers = [], ?string $operationName = null, array $context = []): bool
+    {
+        if (! $data instanceof UserResetPasswordDto) {
+            return false;
+        }
+
+        if ('_api_/user-reset-password_post' !== $operationName) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function process($data, array $identifiers = [], ?string $operationName = null, array $context = [])
+    {
+        if (! $data instanceof UserResetPasswordDto) {
+            throw new \LogicException('$data is not a UserResetPasswordDto object');
+        }
+
+        /** @var UserResetPasswordDto */
+        $userResetPasswordDto = $data;
+
+        switch ($userResetPasswordDto->getEmail()) {
+            case 'user@example.com':
+                return $userResetPasswordDto;
+            case 'does-not-exist@example.com':
+                throw new NotFoundHttpException('User does not exist.');
```

Comment:
An HTTP exception shouldn't be thrown from the outside of the controller layer. Use exception mapping instead. This isn't very important here as we're in the tests, but tests should promote best practices as they are also intended to be used as documentation.

Reply:
done in commit eada6c0bc00762ec50a0c4a2b9889df3bcbf65ae